assets/js/panel.js

```diff
@@ -104,46 +104,15 @@ class Panel {
     this.status.textContent = content;
   }
 
-  loadURL(url, callback) {
-    // load specified url; display article contents in the panel
-    // takes an optional callback; if defined, will be applied to
-    // the url contents before inserting
-
-    // if the requested url is already showing, do nothing
-    if (this.el.dataset.showing && this.el.dataset.showing == url) {
-      return;
+  loadContent(html, callback) {
+    let parser = new DOMParser();
+    const doc = parser.parseFromString(html, "text/html");
+    // Get the article content and insert into panel
+    const article = doc.querySelector("article");
+    if (callback != undefined) {
+      callback(article);
     }
-    // If you need to test load failure, uncomment this
-    // if (Math.random() < 0.5) {
-    //   url = url + "xxx";
-    // }
-    // console.log("fetching:", url);
-
-    fetch(url)
-      .then((response) => {
-        if (!response.ok) {
-          return Promise.reject(response);
-        }
-        return response;
-      })
-      .then((response) => response.text())
-      .then((html) => {
-        let parser = new DOMParser();
-        const doc = parser.parseFromString(html, "text/html");
-        // Get the article content and insert into panel
-        const article = doc.querySelector("article");
-        if (callback != undefined) {
-          callback(article);
-        }
-        this.el.querySelector("article").replaceWith(article);
-        // store loaded url in data attribute for reload check
-        this.el.dataset.showing = url;
-      })
-      .catch((response) => {
-        // if the request failed, display error article
-        let errorArticle = document.querySelector("#loaderror").cloneNode(true);
-        this.el.querySelector("article").replaceWith(errorArticle);
-      });
```

Comment:
We don't need this anymore because we're no longer fetching URLs in the same way? Or are we somehow bypassing fetching?

Reply:
we're bypassing the fetch entirely - I'm including just the html we need in the json data we generate by hugo

Reply:
Clever! nice